File: Diagrammatize.Design.Pattern/code/cxx/Prototype/Sample/MessageBox.cpp
#include "MessageBox.hpp"
#include <iostream>

MessageBox::MessageBox(char decochar_) : decochar(decochar_) {}

MessageBox::~MessageBox() {}

void MessageBox::use(const std::string& s) {
    int length = s.length();
    for (int i = 0; i < length + 4; i++) {
        std::cout << decochar; 
    }
    std::cout << '\n';
    std::cout << decochar << " "  << s << " " << decochar << '\n';
    for (int i = 0; i < length + 4; i++) {
        std::cout << decochar;
    }
    std::cout << '\n';
}

MessageBox* MessageBox::createClone() {
    return new MessageBox(*this);
}

File: Diagrammatize.Design.Patterns/code/python/Prototype/Sample/MessageBox.py
import copy
from framework import Product

class MessageBox(Product):
    def __init__(self, decochar):
        self.decochar = decochar

    def use(self, s):
        length = len(s)
        for i in range(length + 4):
            print(self.decochar, end="")
        print("")
        print(self.decochar + " " + s + " " + self.decochar)
        for i in range(length + 4):
            print(self.decochar, end="")
        print("")

    def createClone(self):
        return copy.deepcopy(self)

if __name__ == "__main__":
    a = MessageBox("*")
    b = a.createClone()
    b.use("abc")
